### DISCUSSION TITLE
[RFC] Select support SelectPanel

### DISCUSSION THREAD

**Discussion by crazyair** [ant-design/ant-design#48689] [RFCs]
# RFC

## Summary

`Select` 支持 `SelectPanel`，可以将下拉框独立展示在当前元素上

## Motivation

用户可以自由控制将 `SelectPanel` 渲染到 `Popover`  `Modal` 等元素上

## Detailed design

```tsx
<Select.SelectPanel />
```

## Demo

https://github.com/react-component/select/pull/1041
![image](https://github.com/ant-design/ant-design/assets/7971419/15deac29-11bc-4c96-9287-0970b6d63d38)

**Comment by afc163** [comment]
Slider 如何？

**Reply by crazyair** [reply]
年龄只是举个例子，还有其他场景的区间，范围比较大，还得用上 Select 的虚拟列表

**Comment by MadCcc** [comment]
用两个 Select 拼一下

**Reply by crazyair** [reply]
可以 HOC 做，但是要一个 Popover 显示 start、end 的 Select，Select 现在还不支持单独渲染下拉框组件的功能。

**Comment by crazyair** [comment]
start、end 是否复用一份 options，还是各自自己的？

**Comment by afc163** [comment]
我觉得这个还是用两个 Select 拼一下比较符合设计需求，很少看到放在一个控件里做的。

**Reply by crazyair** [reply]
看了下 picker，觉得可以支持个 SelectPanel，下拉框直接渲染当前到页面上
